extensions/amp-story-panning-media/0.1/amp-story-panning-media.js

```diff
@@ -32,54 +40,173 @@ export class AmpStoryPanningMedia extends AMP.BaseElement {
+        const imgEl = dev().assertElement(this.element_.querySelector('img'));
         // Remove layout="fill" classes so image is not clipped.
-        this.image_.classList = '';
-        // Fill image to 100% height of viewport.
+        imgEl.classList = '';
+        // Centers the amp-img horizontally. The image does not load if this is done in CSS.
         // TODO(#31515): Handle base zoom of aspect ratio wider than image
-        setStyles(this.image_, {height: '100%'});
-        return this.updatePosition_();
+        setImportantStyles(this.ampImgEl_, {
+          left: 'auto',
+          right: 'auto',
+        });
       })
       .catch(() => user().error(TAG, 'Failed to load the amp-img.'));
   }
 
+  /** @private */
+  initializeListeners_() {
+    this.storeService_.subscribe(
+      StateProperty.CURRENT_PAGE_ID,
+      (currPageId) => {
+        this.isOnActivePage_ = currPageId === this.getPageId_();
+        this.onPageNavigation_();
+      },
+      true /** callToInitialize */
+    );
+    this.storeService_.subscribe(
+      StateProperty.PANNING_MEDIA_STATE,
+      (panningMediaState) => this.onPanningMediaStateChange_(panningMediaState)
+    );
+    // Mutation observer for distance attribute
+    const config = {attributes: true, attributeFilter: ['distance']};
+    const callback = (mutationsList) => {
+      this.distance_ = parseInt(
+        mutationsList[0].target.getAttribute('distance'),
+        10
+      );
+    };
+    const observer = new MutationObserver(callback);
+    this.getPage_() && observer.observe(this.getPage_(), config);
+  }
+
+  /** @private */
+  onPageNavigation_() {
+    if (this.isOnActivePage_) {
+      // TODO(#31932): A key could be sent here to update elements of the same group.
+      // Note, this will not work when there are 2 or more panning components on the same page.
+      // It might need to dynamic to hold more than 1 set of positions.
+      this.animate_();
+    }
+  }
+
+  /**
+   * Animates from current position to active components position.
+   * @private
+   */
+  animate_() {
+    const startPos = this.storeService_.get(
+      StateProperty.PANNING_MEDIA_STATE
+    ) || {x: this.x_, y: this.y_, zoom: this.zoom_};
+
+    const easeInOutQuad = (val) =>
+      val < 0.5 ? 2 * val * val : 1 - Math.pow(-2 * val + 2, 2) / 2;
+
+    let tick = 0;
+    const nextFrame = () => {
+      tick++;
+      if (tick < DURATION) {
+        const easing = easeInOutQuad(tick / DURATION);
+        this.storeService_.dispatch(Action.SET_PANNING_MEDIA_STATE, {
+          x: startPos.x + (this.x_ - startPos.x) * easing,
+          y: startPos.y + (this.y_ - startPos.y) * easing,
+          zoom: startPos.zoom + (this.zoom_ - startPos.zoom) * easing,
+        });
+        this.isOnActivePage_ && requestAnimationFrame(nextFrame);
+      } else {
+        this.storeService_.dispatch(Action.SET_PANNING_MEDIA_STATE, to);
```

Comment:
what is `to` (the second parameter in the dispatch()) here?

Reply:
Reverted, will save for next commit.